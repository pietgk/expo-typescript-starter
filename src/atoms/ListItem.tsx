import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  name: { fontWeight: 'bold' },
  content: {},
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: 'blue',
    marginRight: 8,
  },
});

type Props = {
  onPress: () => void;
  name: string;
  content: string;
};

const ListItem: React.FC<Props> = ({
  onPress,
  name,
  content,
}: Props): JSX.Element => (
  <TouchableOpacity onPress={onPress}>
    <View style={styles.container}>
      <View style={styles.avatar} />
      <View>
        <Text style={styles.name}>{name}</Text>
        <Text>{content}</Text>
      </View>
    </View>
  </TouchableOpacity>
);

export default ListItem;
