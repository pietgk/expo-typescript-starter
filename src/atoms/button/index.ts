export { default as RoundButton } from './RoundButton';
export { default as IconButton } from './IconButton';
