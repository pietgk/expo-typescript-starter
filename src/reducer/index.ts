/* eslint-disable @typescript-eslint/no-unused-vars */
import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

export default combineReducers({
  foo: (state = {}, action: any) => {
    return state;
  },
  bar: persistReducer({ key: 'bar', storage }, (state = {}, action: any) => {
    return state;
  }),
});
