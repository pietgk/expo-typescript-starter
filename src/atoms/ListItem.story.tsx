import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import ListItem from './ListItem';
import { StorybookContainer } from '../templates/StoryContainer';

storiesOf('atoms', module)
  .addDecorator(StorybookContainer)
  .add('ListItem', () => (
    <ListItem
      onPress={action('onPress')}
      name={text('name', 'Bob Dilan')}
      content={text('content', 'hello world')}
    />
  ));
