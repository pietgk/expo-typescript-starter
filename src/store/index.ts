import { createStore, applyMiddleware } from 'redux';
import { persistStore } from 'redux-persist';
import thunk from 'redux-thunk';
import reduxLogger from 'redux-logger';
import reducer from '../reducer';

// development なら redux-loggerを入れる
const logger = __DEV__ ? [reduxLogger] : [];

const middlewares = applyMiddleware(thunk, ...logger);

const initialState = {};

const store = createStore(reducer, initialState, middlewares);
const persistor = persistStore(store);

export { store, persistor };
