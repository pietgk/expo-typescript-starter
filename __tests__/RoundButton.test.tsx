import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { RoundButton } from '../src/atoms';

describe('RoundButton snapshot', () => {
  it('renders the loading screen', async () => {
    const tree = renderer
      .create(<RoundButton label="Hello World!" onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
