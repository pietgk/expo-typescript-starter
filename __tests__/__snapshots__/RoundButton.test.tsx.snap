// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton snapshot renders the loading screen 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "dodgerblue",
      "borderRadius": 100,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    Hello World!
  </Text>
</View>
`;
